### DISCUSSION TITLE
Jacobian (jac_g) makes the optimization not work

### DISCUSSION THREAD

**Discussion by sixtosevenf** [coin-or/Ipopt#718] [Q&A] [answered]
Hi,

I tested my code with the simple example in the document ([https://coin-or.github.io/Ipopt/INTERFACES.html]) also see the below picture.
<img width="425" alt="image" src="https://github.com/coin-or/Ipopt/assets/148242362/6165c53a-0f86-41fc-b35a-1e5009c8c755">

Here the cost function is $x_1x_4(x_1 + x_2 + x_3) + x_3$ and the constraints are $x_1x_2x_3x_4 >= 25$ and $x_1^2 + x_2^2 + x_3^2 + x_4^2 = 40$.
I solved the problem 1) without gradient of cost function and Jacobian of constraint and 2) with gradient of cost function using the option **_grad_f_** and without Jacobian of constraint. However, when I tried solving the problem with Jacobian of constraints using the code **_jac_g_**, the problem is not solved as Infeasible_Problem_Detected. I think the Jacobian of constraint is not wrong since it can be easily derived analytically as follows
```math
J_g = \begin{bmatrix} x_2x_3x_4 & x_1x_3x_4 & x_1x_2x_4 & x_1x_2x_3 \\ 2x_1 & 2x_2 & 2x_3 & 2x_4 \end{bmatrix}
```

Any comments would be appreciated.

**Comment by svigerske** [comment]
Try running the [derivative check](https://coin-or.github.io/Ipopt/SPECIALS.html#DERIVCHECK) and compare your code with the implementation given at https://github.com/coin-or/Ipopt/blob/stable/3.14/examples/hs071_cpp/hs071_nlp.cpp.

Do not enable gradient or Jacobian approximations.

**Reply by sixtosevenf** [reply, marked as answer]
Thanks for your reply and the reference. I had an error in my code. Now I solved the problem.
